src/commentscan.l

```diff
@@ -420,6 +420,7 @@ struct commentscanYY_state
   QCString         guardExpr;
   int              roundCount = 0;
   int              HTMLDetails = 0;
+  std::vector<int> HTMLDetailsSummary;
```

Comment:
Looks like there is an invariant `HTMLDetails==HTMLDetailsSummary.size()`. 
If so `HTMLDetails` can be replaced by `HTMLDetailsSummary.size()`

Reply:
Indeed this is the case, adjusted code.